### PR TITLE
[main] Recreate missing backingFsBlockDev on setting xfs project quota

### PR DESCRIPTION
**Pull request by saschagrunert**
There is the rare case that something removed the backingFsBlockDev, which usually points to `/var/lib/containers/storage/overlay/backingFsBlockDev`. In that case, all `SetQuota` operations would fail on overlay layer creation.

We now work around that error by recreating the backingFsBlockDev.

/hold

I have to test this on CRI-O and rebase the change to c/storage v1.37.0

Refers to https://github.com/containers/storage/pull/1326

### CODE REVIEW

File: drivers/quota/projectquota.go [hunk 1]
```diff
@@ -228,15 +230,24 @@ func setProjectQuota(backingFsBlockDev string, projectID uint32, quota Quota) er
...
-	if errno != 0 {
+
+	// If the backingFsBlockDev does not exist any more then try to recreate it.
+	if errors.Is(errno, unix.ENOENT) {
+		if _, err := makeBackingFsDev(q.basePath); err != nil {
+			return fmt.Errorf(
+				"failed to recreate missing backingFsBlockDev %s for projid %d: %w",
+				q.backingFsBlockDev, projectID, err,
+			)
+		}
```
Comment (nalind) [inline drivers/quota/projectquota.go:247]: I think this is going to need to try the syscall again after `makeBackingFsDev()` succeeds.

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@saschagrunert Whats up with this one?

**Comment by saschagrunert** [comment]
> @saschagrunert Whats up with this one?

The QA needs to verify that it works with CRI-O (see https://issues.redhat.com/browse/OCPBUGS-773)

**Comment by saschagrunert** [comment]
Manually verified that the patch works with CRI-O. Ready for review @rhatdan.

**Comment by rhatdan** [comment]
LGTM

**Review by rhatdan — APPROVED** [review]
@nalind @giuseppe @vrothberg @flouthoc @umohnani8 PTAL

**Pull request merged**